test/test_cli.rb

```diff
@@ -224,6 +224,25 @@ def test_ignore_front_matter
     assert_equal(result[:stdout], expected)
   end
 
+  def test_file_ends_with_single_newline_character_good
```

Comment:
this file is meant to test command-line options, not rules. Can you instead add a test to `tests/rule_tests`? It's really easy you just create a file and add `{MD047}` on the line that should trigger.